include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,350 @@
+                break;
+            case 6:
+                combined = ranges::view::concat(ranges::view::single(frames[0]),
+                                                ranges::view::single(frames[1]),
+                                                ranges::view::single(frames[2]),
+                                                ranges::view::single(frames[3]),
+                                                ranges::view::single(frames[4]),
+                                                ranges::view::single(frames[5]));
+                break;
+            default:
+                throw std::invalid_argument("Invalid number of frames. Choose from 2, 3 or 6.");
+                break;
+        }
+        return combined;
+    }
+
+    /*!\brief A functor that behaves like a named version of std::bind around seqan3::detail::translate_frames_fn::operator().
+     * \tparam size_type Must be an integral type
+     *
+     * \details
+     *
+     * * The single-parameter-operator() of translate_frames_fn would normally return a bound two-parameter-operator() via
+     * std::bind.
+     * * translate_frames_fn's friend operator| would take this as second argument via decltype(std::bind(...)) so that it
+     * is specific to this overload.
+     * * However, since the type of the frame number parameter is a template argument, we need function template
+     * overloading. This doesn't work with decltype() and the actual return type of std::bind is implementation
+     * defined.
+     * * So we have this helper class template which behaves exactly the same, but has a distinct (named) type.
+     *
+     * \attention You should never instantiate this manually.
+     */
+    template <typename size_type>
+    struct delegate
+    {
+        //!\brief The intermediately stored frame number.
+        size_type const num;
+        //!\brief Reference to the parent
+        translate_frames_fn const & parent;
+
+        //!\brief The operator() that only takes the range as argument and forwards to the two-parameter operator().
+        template <typename irng_t>
+        auto operator()(irng_t && irange) const
+        {
+            return parent(std::forward<irng_t>(irange), num);
+        }
+    };
+
+    /*!\brief Range-less interface for use with the pipe notation.
+     * \tparam size_type The type of the input number of frames. Must be an integral type.
+     * \param num Number of frames that should be computed (2, 3 or 6).
+     *
+     * \details
+     * Binds to translate_frames_fn and forwards the number of frames.
+     */
+    template<typename size_type>
+        requires std::is_integral_v<size_type>
+    delegate<size_type> operator()(size_type const num) const
+    {
+        return delegate<size_type>{num, *this};
+        // this doesn't work here, see seqan3::detail::translate_frames_fn::delegate.
+        // return std::bind(translate_frames_fn(), std::placeholders::_1, num);
```

Comment:
Actually for your case std::bind works, that's what I tried to say in my last comments :smile: 
You don't need to make the size_type a template parameter.

In fact I would actually propose making it an enum like in SeqAn2:
http://docs.seqan.de/seqan/2.3.2/enum_TranslationFrames.html

Sorry that I keep coming up with new things :innocent: